[24.2] Fix cancellation of workflow scheduling

ahmedhamidawan opened the pull request:
Fixes https://github.com/galaxyproject/galaxy/issues/19237

Also:
- Fixes `force` router push in `WorkflowRunButton`
- Consolidates styling of `WorkflowNavigationTitle` with the tool form header styling

## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [x] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

ahmedhamidawan committed:
[24.2] Fix cancellation of workflow scheduling

Fixes https://github.com/galaxyproject/galaxy/issues/19237

ahmedhamidawan committed:
fix force routing for `WorkflowRunButton`

and use force prop in `WorkflowNavigationTitle`

ahmedhamidawan committed:
use `to` prop for edit button in `WorkflowNavigationTitle`

This shows a href route for the button in the browser, and keeping an empty `@click` event means we keep the button styling.

ahmedhamidawan committed:
match `WorkflowNavigationTitle` styling with `ToolCard` header

ahmedhamidawan committed:
fix jest to align with the workflow nav header changes

mvdbeek commented on client/src/components/WorkflowInvocationState/WorkflowInvocationState.vue:
```diff
@@ -219,11 +220,22 @@ async function pollJobStatesUntilTerminal() {
+        emit("invocation-cancelled");
+
+        // Update the invocation state to reflect the cancellation
+        setTimeout(async () => {
```
Why the arbitrary timeout here ? I think the responsibility of this method is to to make the request and disable the cancel button. When all of this arrives at the scheduler depends on how busy the workflow scheduler is.

ahmedhamidawan replied:
I just added the fetches (along with a timeout) because without them, this can happen:

https://github.com/user-attachments/assets/70520d79-45c3-471f-98cd-249a65f19992

I.e.: The invocation state is never updated in the store since we stop fetching.

mvdbeek replied:
Well the state of those jobs is still going to change though, no fixed amount of time changes that. You can stop polling the invocation, but not the jobs.

ahmedhamidawan replied:
You're right, and the jobs do poll after this, until `jobStatesTerminal === true`
But the invocation object locally here doesn't update; I'm looking for a way to move the `cancelWorkflowSceduling` function to the invocation store to update invocation there when it is cancelling.

ahmedhamidawan replied:
This next commit 69ae867ae604ee2ed6b5ec6b49c1665cb04377cd fixes this, the fetch is required to update the graph when the state becomes `cancelling`.

mvdbeek replied:
Looks good, thank you!

ahmedhamidawan committed:
poll if invocation state is not terminal again

Also, account for `deleting` state in `useInvocationGraph` when getting the state for a graph step.

ahmedhamidawan committed:
adjust `ProgressBar` props to fix console errors
